app/middlewares/vendor_api_request_middleware.rb

```diff
@@ -0,0 +1,56 @@
+require 'redis'
+require './app/workers/vendor_api_request_worker'
+
+class VendorAPIRequestMiddleware
+  REQUEST_HEADER_KEYS = %w[
+    HTTP_VERSION
+    HTTP_HOST
+    HTTP_USER_AGENT
+    HTTP_ACCEPT
+    HTTP_ACCEPT_ENCODING
+    HTTP_AUTHORIZATION
+    HTTP_CONNECTION
+    HTTP_CACHE_CONTROL
+  ].freeze
+
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    @request = Rack::Request.new(env)
+    status, headers, response = @app.call(env)
+
+    begin
+      if trace_request?
+        VendorAPIRequestWorker.perform_async(request_data, response.body, status, Time.zone.now)
```

Comment:
`response.body` will fail here, as `response` in this context is just an array of output lines, e.g. ` ['<html><body>You are being <a href=\"http://localhost:3000/support/candidates\">redirected</a>.</body></html>']`

Passing `response` as an argument to `perform_async` may still work, though; it's just an array. 

Another concern is the size of enqueued messages, e.g. if response is a json list of applications. I thought we would only store responses if the API call is unsuccessful?

Reply:
Ah yes we need to mitigate this similarly to https://github.com/pdfkit/pdfkit/blob/master/lib/pdfkit/middleware.rb#L24-L25